src/Microsoft.DotNet.Darc/src/Darc/Operations/UpdateDependenciesOperation.cs

```diff
@@ -106,10 +108,15 @@ public override async Task<int> ExecuteAsync()
                     }
 
                     // Start channel query.
-                    var channel = barOnlyRemote.GetChannelAsync(_options.Channel);
+                    Task<Channel> channel = barOnlyRemote.GetChannelAsync(_options.Channel);
 
                     // Limit the number of BAR queries by grabbing the repo URIs and making a hash set.
-                    var repositoryUrisForQuery = dependencies.Select(dependency => dependency.RepoUri).ToHashSet();
+                    // We gather the latest build for any dependencies that aren't marked with coherent parent
+                    // dependencies, as those will be updated based on additional queries.
+                    HashSet<string> repositoryUrisForQuery =
+                        currentDependencies.Where(dependency => string.IsNullOrEmpty(dependency.CoherentParentDependencyName))
+                                           .Select(dependency => dependency.RepoUri).ToHashSet();
```

Comment:
This "align with other stuff" isn't generally the style we use (it tends to make lines really long and have to wrap more).   I think usually these get written like
```csharp
var value = thing
    .Where(...)
    .Select(...)
    .ToHashSet();
```

So it's sort of like reading 3 instructions in a row.